rust/kernel/error.rs

```diff
@@ -104,3 +105,44 @@ impl From<AllocError> for Error {
         Error::ENOMEM
     }
 }
+
+pub fn from_kernel_result_helper<T>(r: Result<T>) -> T
+where
+    T: TryFrom<c_types::c_int>,
+    T::Error: core::fmt::Debug,
+{
+    match r {
+        Ok(v) => v,
+        Err(e) => T::try_from(e.to_kernel_errno()).unwrap(),
+    }
+}
+
+/// Transforms a `crate::error::Result<T>` to a kernel `C` integer result.
```

Comment:
C is not code, i.e.:

```suggestion
/// Transforms a `crate::error::Result<T>` to a kernel C integer result.
```

Does an intra-doc link work here?

```suggestion
/// Transforms a [`crate::error::Result<T>`] to a kernel C integer result.
```